onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3257,10 +3257,65 @@ def aten_matrix_power(self: TensorType, n: int) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_max(self: TensorType) -> TensorType:
+@torch_op("aten::max", trace_only=True)
+def aten_max(self: TReal, dim_or_other: TReal = None, keepdim: BOOL = None) -> TReal:
     """max(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        self = op.Reshape(self, op.Constant(value_int=[-1]))
+
+    output = 1
+
+    if dim_or_other is None and keepdim is None:
+        result = _aten_max_with_no_dim(self)
+    elif keepdim is None:
+        if isinstance(dim_or_other, int):
+            result, indices = _aten_max_with_dim(self, dim_or_other)
+            output = 2
+        else:  # dim_or_other is tensor
```

Comment:
If dim_or_other is a Tensor, then the mapped aten op is maximum. I think we don't need to consider dim_or_other is tensor here.

Reply:
but pytorch have test case for this scenario.

Reply:
I think that test case was designed for torch.max() function. But according to the dim_or_other, it will be mapped to different aten op.

Reply:
Then how to pass the test case? I think it is an overload.